subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultFileOperations.java

```diff
@@ -116,6 +125,19 @@ public FileTree tarTree(Object tarPath) {
         return new FileTreeAdapter(tarTree);
     }
 
+    public <T> Provider<T> defaultProvider(Class<T> clazz) {
```

Comment:
why on this class? doesn't really have anything much to do with file operations.